packages/cli/__tests__/services/template-renderer.test.ts

```
@@ -248,3 +258,207 @@ describe('expandLoopVariables', () => {
+  it('should substitute defined variables', () => {
+    expect(substituteText('Hello {{name}}', { name: 'World' }))
+      .toBe('Hello World');
+  });
+
+  it('should preserve undefined variables as literal text', () => {
+    expect(substituteText('Hello {{name}}', {}))
+      .toBe('Hello {{name}}');
+  });
+
+  it('should apply escape function when provided', () => {
+    const escape = (v: string) => `[${v}]`;
```

Biome (2.3.14)

[error] Do not shadow the global "escape" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)